### ISSUE TITLE
Regression (7.3 -> 7.4): Indexing fails for item IDs that contain slashes

### ISSUE THREAD

**Issue by drunken-monkey** [elastic/elasticsearch-php#1048]
### Summary of problem or feature request

When updating this library from version 7.3 (or lower) to 7.4, indexing (and any other operations where an item ID is included in the URL) starts to fail for items whose IDs contain slashes.

The error printed is the following:
```
PHP Fatal error:  Uncaught Elasticsearch\Common\Exceptions\BadRequest400Exception: {"error":"no handler found for uri [/twitter/_doc/1/1] and method [POST]"} in $PWD/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php:627
Stack trace:
#0 $PWD/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php(312): Elasticsearch\Connections\Connection->process4xxError()
#1 $PWD/vendor/react/promise/src/FulfilledPromise.php(28): Elasticsearch\Connections\Connection->Elasticsearch\Connections\{closure}()
#2 $PWD/vendor/guzzlehttp/ringphp/src/Future/CompletedFutureValue.php(55): React\Promise\FulfilledPromise->then()
#3 $PWD/vendor/guzzlehttp/ringphp/src/Core.php(341): GuzzleHttp\Ring\Future\CompletedFutureValue->then()
 in $PWD/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php on line 627
```

The problem seems to be that the ID (`1/1`) is not URL-encoded, so it’s read as two separate path components.

### Code snippet of problem

With the default `twitter` index set up, just execute this code:

```php
$client->index([
  'index' => 'twitter',
  'id' => '1/1',
  'body' => [],
]);
```
It will succeed without problems when using the client library version 7.1-7.3 (didn’t test with lower) and fail for client library version 7.4. (Installed using: `composer require 'elasticsearch/elasticsearch:7.$i.*'`)

The ES server version used is _not_ relevant. The 7.1 client library works fine (in this example) with both 7.1 and 7.4 servers, but the 7.4 library fails for both server versions. So, it really seems to be a regression in this package, not in the ES server software.

A working (but probably wrong) workaround is to change line 31 of `\Elasticsearch\Endpoints\Index::getURI()`: `$id = $this->id ? urlencode($this->id) : null;`.
But, as said, the same problem occurs for (as far as I could tell) all other endpoints that pass the document ID via the URL (like `Get`), so I’d hope there is a simpler fix than changing this in all affected endpoints manually.

### System details

I’m pretty sure the operating system and PHP version are irrelevant (I also had the problem on AWS), but including them for completeness’ sake:

- Operating System: Arch Linux
- PHP Version: 7.4.7
- ES-PHP client version: 7.4.1
- Elasticsearch version: 7.4.0

**Comment by ezimuel** [comment]
@drunken-monkey thanks for reporting this issue. I'm looking into it and I'll update asap.

**Comment by ezimuel** [comment]
@drunken-monkey this issue is related with https://github.com/elastic/elasticsearch-php/pull/1017. From 7.4 we used a new code generator for the APIs that overwrited the `setId()` in some endpoints witout `urlencode()`. The correct `setId()` implementation is in the [AbstractEndpoint](https://github.com/elastic/elasticsearch-php/blob/7.x/src/Elasticsearch/Endpoints/AbstractEndpoint.php#L163-L176).
This issue has been fixed with `elasticsearch-php` [7.7](https://github.com/elastic/elasticsearch-php/releases/tag/v7.7.0).
Since 7.4, 7.5 and 7.6 are affected by this issue I'm going to release 7.4.2, 7.5.3 and 7.6.2 with this fix.

Thanks for reporting the issue, I'll update when I'll do these patch releases. In the meantime you can try to use 7.7.0.

**Comment by ezimuel** [comment]
@drunken-monkey just released [7.4.2](https://github.com/elastic/elasticsearch-php/releases/tag/v7.4.2). Let me know if this fixes the issue, thanks!

**Comment by drunken-monkey** [comment]
Yes, works perfectly now, thanks!
